### PR TITLE
Rewrite, new features

### PR DESCRIPTION
**Pull request by nwithan8**
Initial commit for EasyPost's derivative of the original Scotch.

- Rewritten from F# to C#
- Added missing headers expected when recreating a request
- Added support for censoring sensitive headers in cassette files
- Replaced XUnit with MSTest for unit tests
- Updated unit tests, removed bad unit tests (will replace in the future)

### CODE REVIEW

File: LICENSE.txt [hunk 1]
```diff
@@ -1,21 +0,0 @@
...
-Copyright (c) 2016 Martin Leech
-
-Permission is hereby granted, free of charge, to any person obtaining a copy
-of this software and associated documentation files (the "Software"), to deal
-in the Software without restriction, including without limitation the rights
-to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
-copies of the Software, and to permit persons to whom the Software is
-furnished to do so, subject to the following conditions:
-
-The above copyright notice and this permission notice shall be included in all
```
Comment (Justintime50) [inline LICENSE.txt:12]: Per the MIT license, we need to maintain the original license even if it's a derivite work.

**Review by Justintime50 — DISMISSED** [review]

File: LICENSE.txt (hunk 1 above)
Comment (nwithan8) [reply LICENSE.txt]: Resolved in [e24f2e1](https://github.com/EasyPost/scotch/pull/1/commits/e24f2e11e88126ae38a5543c9b7226db10764b7e)

File: Scotch.Tests/Scotch.Tests.csproj [hunk 2]
```diff
@@ -1,14 +1,28 @@
 ﻿<?xml version="1.0" encoding="utf-8"?>
 <Project Sdk="Microsoft.NET.Sdk.Web">
   <PropertyGroup>
-    <TargetFramework>netcoreapp2.1</TargetFramework>
+    <TargetFramework>net6.0</TargetFramework>
```
Comment (Justintime50) [inline Scotch.Tests/Scotch.Tests.csproj:4]: Is this backwards compatible with net5.0? Should we be targeting both?

File: Scotch.Tests/Scotch.Tests.csproj [hunk 3]
```diff
@@ -1,14 +1,28 @@
...
     <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
   </PropertyGroup>
   <ItemGroup>
-    <ProjectReference Include="..\scotch\Scotch.fsproj" />
+    <ProjectReference Include="..\Scotch\Scotch.csproj" />
+  </ItemGroup>
+  <ItemGroup>
+    <PackageReference Include="MSTest.TestAdapter" Version="2.2.8" />
+    <PackageReference Include="MSTest.TestFramework" Version="2.2.8" />
+    <PackageReference Remove="FSharp.Core" />
```
Comment (Justintime50) [inline Scotch.Tests/Scotch.Tests.csproj:16]: Is this needed now that it's C# only?

File: Scotch/Tools.cs [hunk 4]
```diff
@@ -0,0 +1,21 @@
+using System.Runtime.CompilerServices;
```
Comment (Justintime50) [inline Scotch/Tools.cs:1]: `Utils` may make more sense as a filename?

**Review by Justintime50 — CHANGES_REQUESTED** [review]
Code looks good aside from a couple nit questions.

After some reading, can we please add the following directly under the already present copyright notice in the license file: `Copyright (c) 2022 EasyPost (Simpler Postage, Inc)`.

Finally, can you update the README calling out the following things:
- This is a fork of the original repo with a one/two liner explaining why we forked it/what it does different
- Update the rest of the README as needed to modernize the language based on what's changed and/or to rebrand it as EasyPost
- A simple minimalistic example on how to use this lib (with the new syntax) so that if needed someone stumbling on it prior to release has enough info to go on (AKA: ourselves even)

File: Scotch.Tests/Scotch.Tests.csproj (hunk 2 above)
Comment (nwithan8) [reply Scotch.Tests/Scotch.Tests.csproj]: Resolved in [fb7f153](https://github.com/EasyPost/scotch/pull/1/commits/fb7f15380aa31ddb1f7b2ab77e2b4677e1d5a0d1)

File: Scotch.Tests/Scotch.Tests.csproj (hunk 3 above)
Comment (nwithan8) [reply Scotch.Tests/Scotch.Tests.csproj]: Resolved in [fb7f153](https://github.com/EasyPost/scotch/pull/1/commits/fb7f15380aa31ddb1f7b2ab77e2b4677e1d5a0d1)

File: Scotch/Tools.cs (hunk 4 above)
Comment (nwithan8) [reply Scotch/Tools.cs]: Resolved in [ad78b60](https://github.com/EasyPost/scotch/pull/1/commits/ad78b60d7fb4845421e9b717c6e3bb519050be06)

**Review by Justintime50 — APPROVED** [review]

**Pull request merged**